collector/mesos_agent.go

```diff
@@ -137,7 +137,7 @@ func (h *DCOSHost) getAgentState() (agentState, error) {
 	c := NewHTTPClient(
 		strings.Join([]string{h.IPAddress, strconv.Itoa(h.Port)}, ":"),
 		"/state",
-		time.Duration(15*time.Second))
+		time.Duration(2*time.Second))
```

Comment:
It's enough, and blocks gathering metrics. Remember, we're getting this from the localhost, so anything more than 2 seconds and something is wrong.